test/functional/sessions.test.js

```diff
@@ -216,5 +217,22 @@ describe('Sessions', function () {
           });
       })
     });
+    it('should throw error with explicit session', {
+      metadata: { requires: { topology: 'single' } },
+      test: withMonitoredClient('insert', { clientOptions: { w: 0 } }, function (
+        client,
+        events,
+        done
+      ) {
+        const session = client.startSession({ causalConsistency: true });
+        client
+          .db('test')
+          .collection('foo')
+          .insertOne({ foo: 'bar' }, { session }, err => {
+            expect(err).to.exist;
```

Comment:
We have a well-defined message, let's test that it matches here

Reply:
sounds good